examples/DatePicker.basic.jsx

```diff
@@ -3,16 +3,39 @@
  * See LICENSE.md in the project root for license terms and full copyright notice.
  *--------------------------------------------------------------------------------------------*/
 import * as React from 'react';
-import { DatePicker } from '@itwin/itwinui-react';
+import { DatePicker, IconButton, Popover } from '@itwin/itwinui-react';
+import { SvgCalendar } from '@itwin/itwinui-icons-react';
 
 export default () => {
   const [currentDate, setCurrentDate] = React.useState(new Date());
+  const [visible, setVisible] = React.useState(false);
   return (
-    <DatePicker
-      date={currentDate}
-      onChange={(date) => {
-        setCurrentDate(date);
-      }}
-    />
+    <>
+      <Popover
+        content={
+          <DatePicker
+            date={currentDate}
+            onChange={(date) => {
+              setCurrentDate(date);
+              setVisible(false);
+            }}
+            setFocus
+          />
+        }
+        placement='bottom-start'
+        visible={visible}
+        onVisibleChange={setVisible}
+      >
+        <IconButton
+          label='Choose date'
+          onClick={() => {
+            setVisible(!visible);
+          }}
+        >
+          <SvgCalendar />
+        </IconButton>
+      </Popover>
+      <span>{currentDate.toString()}</span>
```

Comment:
Prefer `toDateString()` over `toString()` to make it less verbose, at least for the examples where we do not enable time selection.

Also, there should probably be a wrapping `<div>`, because now the empty space looks weird.

Current:
![image](https://github.com/iTwin/iTwinUI/assets/9084735/73ebbd27-1dc7-4dde-91cf-68ccf799ae35)

Expected:
![image](https://github.com/iTwin/iTwinUI/assets/9084735/9bdc6ddd-a29a-40b0-833e-22e333327e93)

Reply:
Will do.

Reply:
Done in latest commit.

Reply:
Prefer flex `gap` over margin, and prefer variables over magic numbers.

Reply:
Changed in latest commit.